app/sliders/importer.py

```diff
@@ -166,6 +174,48 @@ def init_nested_dict(self):
         :return: defaultdict, a nested dictionary
         """
         return defaultdict(self.init_nested_dict)
+    
+
+class FUEDataParser:
+    def parse(self, sheets):
+        data = self.extract(sheets)
+
+        return data
+    
+    def extract(self, sheets):
+        """
+        Extract Fue Expenses data.
+
+        :param sheets: dict, where keys are names of sheets and values are pandas DataFrames
+        :return: nested dict, with the following shape {area: {year: {x: unit[], y: value[]}}}
+        """
+        extracted = self.init_nested_dict()
+        relevant_years = [2011, 2020]
+        for sheet_key in sheets:
+            basis, _, area = sheet_key.split("_")
+            if basis == "fue":
+                df = sheets[sheet_key]
+                row = df.loc[df["Jahr"].isin(relevant_years)]
```

Comment:
This line of code is making a subset of the dataframe that contains all the rows whose year is part of the relevant_years list. Some lines below, you are doing `row.iloc[0][unit]`, where you take the first row (with index 0). This means that you will always take the value for the first year. But we want to extract the value for each year.

I suggest doing the following changes:

- Delete the line `row = df.loc[df["Jahr"].isin(relevant_years)]`.
- Instead of `value = row.iloc[0][unit]`, do:

```
row = df.loc[df["Jahr"] == year]
value = row.iloc[0][unit]
```

Reply:
yes, neat. i changed it